File: Data structures/C#/Basic/Queue.cs
public class Queue
{
    private Queue<int> queue = new Queue<int>();

    public void Enqueue(int value)
    {
        queue.Enqueue(value);
    }

    public int Dequeue()
    {
        if (queue.Count == 0) throw new InvalidOperationException("Queue is empty");
        return queue.Dequeue();
    }

    public int Peek()
    {
        if (queue.Count == 0) throw new InvalidOperationException("Queue is empty");
        return queue.Peek();
    }
}

File: Learn-Datastructures/C/Queue.c
#include <stdio.h>
#define MAX 5

int queue[MAX];
int front = -1, rear = -1;

void enqueue(int value) {
    if (rear == MAX - 1)
        printf("Queue overflow\n");
    else {
        if (front == -1) front = 0;
        queue[++rear] = value;
    }
}

int dequeue() {
    if (front == -1 || front > rear)
        printf("Queue underflow\n");
    else
        return queue[front++];
}

void display() {
    if (front == -1 || front > rear)
        printf("Queue is empty\n");
    else {
        for (int i = front; i <= rear; i++)
            printf("%d ", queue[i]);
        printf("\n");
    }
}

int main() {
    enqueue(10);
    enqueue(20);
    enqueue(30);
    display();
    dequeue();
    display();
    return 0;
}
